cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/replication/BasicPullStrategyTest.java

```diff
@@ -403,4 +408,82 @@ public void pull_indexesUpdated()
 
     }
 
+    /**
+     * Test that a replication can be stopped.
+     * Uses a lot of large-ish docs to ensure the replication takes some time.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void stopRunningReplication() throws Exception {
+        // Create a lot of documents in the remote db
+        int numDocs = 5000;
+        Random r = new Random();
+        for (int i = 0; i < numDocs; i++) {
+            Foo f = new Foo();
+            f.setId(i + CouchUtils.generateDocumentId());
+            f.setRevision(CouchUtils.getFirstRevisionId());
+            byte[] bytes = new byte[512];
+            r.nextBytes(bytes);
+            f.setFoo("Foo " + i + " " + new String(bytes, "UTF-8"));
+            remoteDb.create(f);
+        }
+
+        // Start the replication
+        Replicator replicator = getPullBuilder().build();
+        replicator.start();
+
+        // Wait until the replicator is already started before stopping
+        do {
+            TimeUnit.MILLISECONDS.sleep(100);
+        } while (replicator.getState() == Replicator.State.PENDING);
+
+        // Assert that the replicator is started
+        Assert.assertEquals("The replicator should be started", Replicator.State.STARTED,
+                replicator.getState());
+
+        // Let the replicator run for a short time
+        TimeUnit.MILLISECONDS.sleep(500);
```

Comment:
Added an assertion in 294cfd5.
Also reduced the overall number of docs and wrote them in bulk batches to improve test performance (from ~22 s to ~ 4 s).